src/auth/src/credentials/service_account.rs

```diff
@@ -12,53 +12,242 @@
+pub(crate) fn creds_from(js: Value) -> Result<Credential> {
+    Builder::new(js).build()
+}
 
-    Ok(Credential {
-        inner: Arc::new(ServiceAccountCredential { token_provider }),
-    })
+#[derive(Debug)]
+enum ServiceAccountRestrictions {
+    Audience(String),
+    Scopes(Vec<String>),
 }
 
-/// A representation of a Service Account File. See [Service Account Keys](https://google.aip.dev/auth/4112)
-/// for more details.
-#[derive(serde::Deserialize, Builder)]
-#[builder(setter(into))]
-struct ServiceAccountInfo {
+impl ServiceAccountRestrictions {
+    fn get_audience(&self) -> Option<&String> {
+        match self {
+            ServiceAccountRestrictions::Audience(aud) => Some(aud),
+            ServiceAccountRestrictions::Scopes(_) => None,
+        }
+    }
+
+    fn get_scopes(&self) -> Option<&[String]> {
+        match self {
+            ServiceAccountRestrictions::Scopes(scopes) => Some(scopes),
+            ServiceAccountRestrictions::Audience(_) => None,
+        }
+    }
+}
+
+/// A builder for constructing service account [Credential] instances.
+///
+/// # Example
+/// ```
+/// # use google_cloud_auth::credentials::service_account::Builder;
+/// # tokio_test::block_on(async {
+/// let key = serde_json::json!({
+/// "client_email": "test-client-email",
+/// "private_key_id": "test-private-key-id",
+/// "private_key": "<YOUR_PKCS8_PEM_KEY_HERE>",
+/// "project_id": "test-project-id",
+/// "universe_domain": "test-universe-domain",
+/// });
+/// let credentials = Builder::new(key).with_aud("https://pubsub.googleapis.com").build();
+/// })
+/// ```
+pub struct Builder {
+    service_account_key: Value,
+    restrictions: ServiceAccountRestrictions,
+    quota_project_id: Option<String>,
+}
+
+impl Builder {
+    /// Creates a new builder using [service_account_key] JSON value.
+    /// By default, the builder is configured with [cloud-platform] scope.
+    /// This can be overridden using [with_aud][Builder::with_aud]
+    /// or [with_scopes][Builder::with_scopes] methods.
+    ///
+    /// [cloud-platform]:https://cloud.google.com/compute/docs/access/service-accounts#scopes_best_practice
+    /// [service_account_key]: https://cloud.google.com/iam/docs/keys-create-delete#creating
+    pub fn new(service_account_key: Value) -> Self {
+        Self {
+            service_account_key,
+            restrictions: ServiceAccountRestrictions::Scopes(
+                DEFAULT_SCOPES.into_iter().map(|s| s.into()).collect(),
+            ),
+            quota_project_id: None,
+        }
+    }
+
+    /// Sets the audience for this credential.
+    ///
+    /// `aud` is a [JWT] claim specifying intended recipient(s) of the token,
+    /// that is, a service(s).
+    /// Only one of audience or scopes can be specified for a credential.
+    /// Setting the audience will replace any previously configured scopes.
+    /// The value should be https://{SERVICE}/. (e.g. `https://pubsub.googleapis.com/`)
+    ///
```

Comment:
Consider a rustdoc example:

```suggestion
    ///
    /// # Example
    /// ```
    /// # use google_cloud_auth::credentials::service_account::Builder;
    /// fn example(service_account_key: &str) -> Result<Credential> {
    ///     Builder::new(service_account_key).with_aud("https://bigtable.googleapis.com/").build()
    /// }
```

or 

```
    /// ```
    /// # use google_cloud_auth::credentials::service_account::Builder;
    /// let service_account_key = "{ /* add details here */ }";
    /// let credentials = Builder::new(service_account_key).with_aud("https://bigtable.googleapis.com/").build()
```